Content.Client/UserInterface/Controls/RadialContainer.xaml.cs

```diff
@@ -0,0 +1,441 @@
+
+        CloseButton.Controller.OnMouseEntered += (_) =>
+        {
+            PlaySizeAnimation(CloseButton, _focusSize, OutSizeAnimationKey, InSizeAnimationKey);
+            ActionLabel.SetMarkup(CloseButton.Content ?? string.Empty);
+            ActionLabel.Visible = true;
+        };
+
+        CloseButton.Controller.OnMouseExited += (_) =>
+        {
+            PlaySizeAnimation(CloseButton, _normalSize, InSizeAnimationKey, OutSizeAnimationKey);
+            ActionLabel.Visible = false;
+        };
+        CloseButton.Controller.OnPressed += (_) =>
+        {
+            Close();
+        };
+    }
+
+    private void PlayRadialAnimation(Control button, Vector2 pos, string playKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(_moveAniTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.Offset),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(new Vector2(0,0), 0f),
+                        new AnimationTrackProperty.KeyFrame(pos, _moveAniTime)
+                    }
+                }
+            }
+        };
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    private void PlaySizeAnimation(Control button, Vector2 size, string playKey, string? stopKey)
+    {
+        var anim = new Animation
+        {
+            Length = TimeSpan.FromMilliseconds(_focusAniTime * 1000),
+            AnimationTracks =
+            {
+                new AnimationTrackControlProperty
+                {
+                    Property = nameof(RadialItem.ButtonSize),
+                    InterpolationMode = AnimationInterpolationMode.Linear,
+                    KeyFrames =
+                    {
+                        new AnimationTrackProperty.KeyFrame(button.Size, 0f),
+                        new AnimationTrackProperty.KeyFrame(size, _focusAniTime)
+                    }
+                }
+            }
+        };
+
+        if (stopKey != null && button.HasRunningAnimation(stopKey))
+            button.StopAnimation(stopKey);
+        if (!button.HasRunningAnimation(playKey))
+            button.PlayAnimation(anim, playKey);
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        base.Draw(handle);
+
+        foreach (var child in Layout.Children)
+        {
+            var button = (RadialItem)child;
+            LayoutContainer.SetPosition(child, button.Offset - (button.Size/2));
+        }
+        LayoutContainer.SetPosition(CloseButton, CloseButton.Offset - (CloseButton.Size/2));
+        LayoutContainer.SetPosition(ActionLabel,
+            new Vector2(0 - (GetTextWidth(ActionLabel) / 4), GetDistance(4.5f) ));
+    }
```

Comment:
Why do controls need their position adjusted every frame?

Reply:
Well, it used for sizing animation. We need to centralize RadialItems in their position, when size was animate.